RealmSwift/Tests/ThreadSafeReferenceTests.swift

```diff
@@ -330,3 +330,480 @@ class ThreadSafeReferenceTests: TestCase {
+        XCTAssertEqual("D", testStruct.arcResults![1].name)
+        XCTAssertEqual("B", testStruct.arcResults![2].name)
+        XCTAssertEqual(4, testStruct.arcList!.count)
+        XCTAssertEqual("A", testStruct.arcList![0].name)
+        XCTAssertEqual("B", testStruct.arcList![1].name)
+        XCTAssertEqual("C", testStruct.arcList![2].name)
+        XCTAssertEqual("D", testStruct.arcList![3].name)
+        XCTAssertEqual(4, testStruct.arcSet!.count)
+        assertAnyRealmCollectionContains(testStruct.arcSet!, keyPath: \.name, items: ["A", "B", "C", "D"])
+
+        dispatchSyncNewThread {
+            XCTAssertEqual(6, testStruct.arcResults!.count)
+            XCTAssertEqual("D", testStruct.arcResults![0].name)
+            XCTAssertEqual("D", testStruct.arcResults![1].name)
+            XCTAssertEqual("B", testStruct.arcResults![2].name)
+            XCTAssertEqual(4, testStruct.arcList!.count)
+            XCTAssertEqual("A", testStruct.arcList![0].name)
+            XCTAssertEqual("B", testStruct.arcList![1].name)
+            XCTAssertEqual("C", testStruct.arcList![2].name)
+            XCTAssertEqual("D", testStruct.arcList![3].name)
+            XCTAssertEqual(4, testStruct.arcSet!.count)
+            self.assertAnyRealmCollectionContains(testStruct.arcSet!, keyPath: \.name, items: ["A", "B", "C", "D"])
+        }
+    }
+}
+
+#if swift(>=5.5)
+extension ThreadSafeWrapperTests {
+    func testThreadSafeWrapperInline() throws {
+        let values = ["A", "B", "C", "D"]
+        try autoreleasepool {
+            let realm = try Realm()
+            try realm.write {
+                realm.create(SwiftStringObject.self, value: ["A"])
+                realm.create(SwiftStringObject.self, value: ["B"])
+                realm.create(SwiftStringObject.self, value: ["C"])
+                realm.create(SwiftStringObject.self, value: ["D"])
+            }
+        }
+
+        let realm = try! Realm()
+        @ThreadSafe var results = realm.objects(SwiftStringObject.self)
+        dispatchSyncNewThread {
+            guard let results = results else {
+                return XCTFail("no results")
+            }
+            results.indices.forEach { idx in
+                XCTAssertEqual(results[idx].stringCol, values[idx])
+            }
+        }
+        @ThreadSafe var swiftStringObject = results!.first
+        dispatchSyncNewThread {
+            guard let swiftStringObject = swiftStringObject else {
+                return XCTFail("no results")
+            }
+
+            XCTAssertEqual(swiftStringObject.stringCol, "A")
+        }
+    }
+
+    func mutateStringCol(@ThreadSafe stringObj: SwiftStringObject?) {
+        guard let stringObj = stringObj else {
+            XCTFail("no object found")
+            return
+        }
+        try! stringObj.realm!.write {
+            stringObj.stringCol = "after"
+        }
+    }
+
+    func testThreadSafeFunctionArgument() {
```

Comment:
This test would pass even if mutateStringCol()'s argument wasn't `@ThreadSafe`. The dispatch should be happening inside the function using `@ThreadSafe`, not the caller.